### PR TITLE
Homework 4 solutions

### PR DESCRIPTION
**Pull request by arizvo**

### CODE REVIEW

File: src/Task1c.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Scanner reader = new Scanner(System.in);
+        Integer total = null;
+        int exitLoopCondition = -1000;
+        int newNumber;
+
+        do {
+            if (total == null) {
+                total = 0;
+            }
+            System.out.print("Please enter integer: ");
```
Comment (gorjanK) [inline src/Task1c.java]: why didnt you use \n as a new line?

File: src/Task1d.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        } while (newNumber != exitLoopCondition);
+    }
+
+    public static void writeResult (int numberA, int numberB, int finalResult) {
+        System.out.print("Total of " + numberA);
+        System.out.print(" and " + numberB);
+        System.out.println(" is " + finalResult);
+    }
+
+    public static int calculateResult (int numberA, int numberB, String arithmeticOperator) {
```
Comment (gorjanK) [inline src/Task1d.java]: why not use char , char is a primitive type you initialize a string object just for one char.

File: src/Task2.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        do {
+            System.out.println("Enter numbers (enter -10 to exit): ");
+            do {
+                newNumber = reader.nextInt();
+                if (newNumber != exitLoopCondition) {
+                    arrayElements += (newNumber + ", ");
+                }
+            } while (newNumber != exitLoopCondition);
+
+            System.out.println(arrayElements.substring(0, arrayElements.length() - 2));
```
Comment (gorjanK) [inline src/Task2.java]: nismo bas koristili arrayElements :D imam osjecaj da je cheeting malo :D

File: src/Task3.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        System.out.println("Task 3: ");
+        System.out.println("Menu: ");
+        System.out.println("1. Is string a palindrome");
+        System.out.println("2. Filter an Array");
+        System.out.println("3. Number is divisible by X.");
+        System.out.println("\nChoose option (1, 2 or 3): ");
+        int menuSelector = reader.nextInt();
+        reader.nextLine(); // "swallow" leftover newline
+
+        if (menuSelector == 1) {
```
Comment (gorjanK) [inline src/Task3.java]: You have different logics within ifs each one should have invoked function which deals with those specific logics, it keeps the code clean and it keeps functions hold all logics that you need to use for that specific case.

File: src/Task2.java (hunk 3 above)
Comment (arizvo) [reply src/Task2.java]: arrayElements je u ovom slucaju ime varijable tipa String, jesi li mozda mislio na funkciju substring? Mislio sam da je dozvoljeno koristiti metode za String ali mogu uraditi drugacije

Comment (arizvo) [reply src/Task2.java]: ispravio sam tako da se ne koristi substring

**Pull request merged**